examples/echoservice/main.go

```diff
@@ -47,3 +51,24 @@ func main() {
 	<-ctx.Done()
 
 }
+
+func setupSignalHandlers(nc *nats.Conn) {
+	go func() {
+		signal.Reset(syscall.SIGINT, syscall.SIGTERM, syscall.SIGUSR1, syscall.SIGUSR2, syscall.SIGHUP)
+		c := make(chan os.Signal, 1)
+		signal.Notify(c, os.Interrupt, syscall.SIGTERM, syscall.SIGINT, syscall.SIGQUIT)
+
+		for {
+			switch s := <-c; {
+			case s == syscall.SIGTERM || s == os.Interrupt || s == syscall.SIGQUIT:
+				slog.Info("Caught signal, requesting clean shutdown", slog.String("signal", s.String()))
+				nc.Drain()
+				os.Exit(0)
```

Comment:
~Is there any other option than `os.Exit`?~
Meh, i just noticed this is the echo example...should be fine